src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -1,16 +1,44 @@
+import { BigNumber } from '@ethersproject/bignumber'
 import { useCallback } from 'react'
 import { useClearQuote, useUpdateQuote } from 'state/price/hooks'
 import { getCanonicalMarket, registerOnWindow } from 'utils/misc'
 import { FeeQuoteParams, getFeeQuote, getPriceQuote } from 'utils/operator'
+import { FeeInformation, PriceInformation } from '../state/price/reducer'
 
 export interface RefetchQuoteCallbackParmams {
   quoteParams: FeeQuoteParams
   fetchFee: boolean
+  previousFee?: FeeInformation
 }
 
-function handleError(err: any): undefined {
-  console.error('Error fetching price/fee', err)
-  return undefined
+async function getQuote({
+  quoteParams,
+  fetchFee,
+  previousFee
+}: RefetchQuoteCallbackParmams): Promise<[PriceInformation, FeeInformation]> {
+  const { sellToken, buyToken, amount, kind, chainId } = quoteParams
+  const { baseToken, quoteToken } = getCanonicalMarket({ sellToken, buyToken, kind })
+
+  // Get a new fee quote (if required)
+  const feePromise =
+    fetchFee || !previousFee ? getFeeQuote({ chainId, sellToken, buyToken, amount, kind }) : previousFee
+
+  // Get a new price quote
+  let exchangeAmount
+  if (kind === 'sell') {
+    // Sell orders need to deduct the fee from the swapped amount
+    exchangeAmount = BigNumber.from(amount)
+      .sub((await feePromise).amount)
+      .toString()
+  } else {
+    // For buy orders, we swap the whole amount, then we add the fee on top
+    exchangeAmount = amount
+  }
+
+  // Get price for price estimation
+  const pricePromise = getPriceQuote({ chainId, baseToken, quoteToken, amount: exchangeAmount, kind })
```

Comment:
i see, so we call it again to get the proper price using the fee adjusted input. hmm yeah it will be nice to have them consolidated with the price already returned using the fee adjusted amount because this is strange